crates/solver/src/driver.rs

```diff
@@ -435,13 +452,34 @@ impl Driver {
                     .map_err(|err| anyhow!("{err}"))
                     .context("convert nonce")?,
             };
+            let scores = model::solver_competition::Scores {
+                winner: winning_solver.account().address(),
+                winning_score: scores.next().expect("no score"), // guaranteed to exist
+                // reference score is the second highest score, or 0 if there is only one score (see
+                // CIP20)
+                reference_score: scores.next().unwrap_or(0.into()),
+                block_deadline: {
+                    let deadline = self.solver_time_limit
+                        + self.solution_submitter.max_confirm_time
+                        + self.additional_mining_deadline;
+                    let number_blocks = deadline.as_secs() / self.block_time.as_secs();
+                    block_during_simulation + number_blocks
+                },
+            };
+            let participants = solver_competition
+                .solutions
+                .iter()
+                .map(|solution| solution.solver_address)
+                .collect();
```

Comment:
This slipped through the cracks earlier but currently Gnosis solvers can return multiple solutions which means they would show up multiple times in the participant list (which is probably not intended). An easy fix would be to make `participants` a `HashSet` in the `model::solver_competition::Request`.

Reply:
Would it then be reasonable to collect as a set and then cast back to vector (or whatever it is supposed to be).

Reply:
nit: would be simpler to just make the `participants` a `HashSet` in the solver competition data upload. Then `.collect()` would already collect into the right thing. But feel free to merge already due to urgency.